fineract-loan/src/main/java/org/apache/fineract/portfolio/interestpauses/api/LoanInterestPauseApiResource.java

```diff
@@ -0,0 +1,129 @@
+
+        return this.toApiJsonSerializer.serialize(result);
+    }
+
+    @POST
+    @Path("/external-id/{loanExternalId}/interest-pauses")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Create a new interest pause for a loan using external ID", description = "Allows users to define a period during which no interest will be accrued for a specific loan using the external loan ID.")
+    @ApiResponses({ @ApiResponse(responseCode = "200", description = "OK") })
+    public String createInterestPauseByExternalId(
+            @PathParam("loanExternalId") @Parameter(description = "loanExternalId") final String loanExternalId,
+            @RequestBody(required = true) final InterestPauseRequestDto request) {
+
+        this.context.authenticatedUser().validateHasReadPermission(RESOURCE_NAME_FOR_PERMISSIONS);
+
+        final CommandWrapper commandRequest = new CommandWrapperBuilder().createInterestPauseByExternalId(loanExternalId)
+                .withJson(request.toJson()).build();
+
+        final CommandProcessingResult result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
+
+        return this.toApiJsonSerializer.serialize(result);
+    }
+
+    @GET
+    @Path("/{loanId}/interest-pauses")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Retrieve all interest pause periods for a loan", description = "Fetches a list of all active interest pause periods for a specific loan.")
+    @ApiResponses({ @ApiResponse(responseCode = "200", description = "OK") })
+    public List<InterestPauseResponseDto> retrieveInterestPauses(
+            @PathParam("loanId") @Parameter(description = "loanId") final Long loanId) {
+
+        this.context.authenticatedUser().validateHasReadPermission(RESOURCE_NAME_FOR_PERMISSIONS);
+
+        return this.interestPauseReadPlatformService.retrieveInterestPauses(loanId).stream()
+                .map(data -> new InterestPauseResponseDto(data.getId(), data.getStartDate(), data.getEndDate(), data.getDateFormat(),
+                        data.getLocale()))
+                .toList();
+    }
+
+    @GET
+    @Path("/external-id/{loanExternalId}/interest-pauses")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Retrieve all interest pause periods for a loan using external ID", description = "Fetches a list of all active interest pause periods for a specific loan using the external loan ID.")
+    @ApiResponses({ @ApiResponse(responseCode = "200", description = "OK") })
+    public List<InterestPauseResponseDto> retrieveInterestPausesByExternalId(
+            @PathParam("loanExternalId") @Parameter(description = "loanExternalId") final String loanExternalId) {
+
+        this.context.authenticatedUser().validateHasReadPermission(RESOURCE_NAME_FOR_PERMISSIONS);
+
+        return this.interestPauseReadPlatformService.retrieveInterestPausesByExternalId(loanExternalId).stream()
```

Comment:
I dont need we need to map `InterestPauseResponseDto` to `InterestPauseResponseDto` :)

Reply:
Fixed